app_architecture/ruby/ruby_on_rails.adoc

```diff
@@ -0,0 +1,332 @@
+* Create a *route* that will serve applications root page with the created *controller* and *view*  
+
+Rails offers a generator that will do all this necessary steps for you. Just type:
+
+----
+$ rails generate controller welcome index
+----
+
+All the necessary files have been created, now we just need to edit line 2 in `config/routes.rb`
+file to look like:
+----
+root 'welcome#index'
+---- 
+
+Run the rails server to verify the page is available
+
+----
+$ rails server
+----
+
+You should see your page by visiting link:http://localhost:3000[http://localhost:3000] in our browser.
+I case you don't see the page, check the link:TBD[logs] that are output to your server to debug.
+
+[[store-your-app-in-git]]
+
+=== Store your App in Git
+
+OpenShift relies on link:http://git-scm.com/[git] , a distributed source control management tool, for
+deploying your project. If your project is not already in `git` first verify that git is on your system:
+
+----
+$ git --help
+usage: git [--version] [--help] [-C <path>] [-c name=value]
+           [--exec-path[=<path>]] [--html-path] [--man-path] [--info-path]
+           [-p|--paginate|--no-pager] [--no-replace-objects] [--bare]
+           [--git-dir=<path>] [--work-tree=<path>] [--namespace=<name>]
+           <command> [<args>]
+----
+
+I case you don't get `command not found` you will need to install it on your host system.
+
+Once you've verified that git works, first make sure you are in your Rails app directory by running
+`ls -1` command. The output of the command should look like:
+
+----
+$ ls -1
+app
+bin
+config
+config.ru
+db
+Gemfile
+Gemfile.lock
+lib
+log
+public
+Rakefile
+README.rdoc
+test
+tmp
+vendor
+----
+
+Now run these commands in your Rails app directory to initialize and commit your code to git:
+
+----
+$ git init
+$ git add .
+$ git commit -m "initial commit"
+----
+
+After your application is committed, set your remote link:https://help.github.com/articles/creating-a-new-repository/[github repository]:
+
+----
+$ git remote add origin git@github.com:<namespace/repository-name>.git
+----
+
+After that, push your application to your remote git repository.
+
+----
+$ git push
+----
+
+== Deploy your application to OpenShift
+
+Deploying your application in OpenShift will be done in three steps:
+
+* Creating frontend service from OpenShift link:../../using_images/sti_images/ruby.html[Ruby 2.0 builder image] and your Ruby on Rails source code
+* Creating database service from OpenShift link:../../using_images/db_images/postgresql.html[PostgreSQL image]
+* Wiring the frontend service to communicate with database
+
+=== Creating frontend service
+
+[WARNING]
+====
+*ToBeDone*:
+We in OpenShift need a way how specify deployment hooks upon the deployment process. The use case in the application is the set the databse migration(and maybe also the DB seeding) upon the deployment. For now the hooks can be specified only in the template.
```

Comment:
@jhadvig 
for now i have another suggestion:

just tell them to rsh into the running application pod and execute the migration.sh script, after the application deploys, and mention that if they wanted to, they could setup a deployment hook to do that for them.  (And that they'll need to redo that step if they recreate their database)

what do you think?